src/ir_Bosch.cpp

```diff
@@ -99,11 +99,10 @@ uint8_t* IRBosch144AC::getRaw(void) {
 /// Set the internal state from a valid code for this protocol.
 /// @param[in] new_code A valid code for this protocol.
 void IRBosch144AC::setRaw(const uint8_t new_code[]) {
-  if (sizeof(new_code) == kBosch144StateLength) {
-    setPower(true);
-    std::memcpy(_.raw, new_code, kBosch144StateLength);
+  if (new_code == kBosch144Off) {
```

Comment:
This doesn't do what you think it does.
It's comparing if the two pointers are the same, not the contents of the two arrays (where the pointers point to) are the same. i.e. You're checking if the two arrays have the same memory address. They won't.
You need to use [`std::memcmp()`](https://en.cppreference.com/w/cpp/string/byte/memcmp) to compare the contents.

You will also need to pass in the `length` of the array as a parameter to `setRaw()` and make sure you don't exceed the size of `_.raw`

Something like:
```c++
/// Set the internal state from a valid code for this protocol.
/// @param[in] new_code A valid code for this protocol.
/// @param[in] length Size of the array being passed in in bytes.
void IRBosch144AC::setRaw(const uint8_t new_code[], const uint16_t length) {
  const uint16_t len = min(length, kBosch144StateLength);
  if (std::memcmp(kBosch144Off, new_code, min(sizeof(kBosch144Off), len))) == 0) // Is it an off message?
    setPower(false);  // It is.
  else
    setPower(true);
  std::memcpy(_.raw, new_code, len);
}
```
and in `ir_Bosch.h`:
```c++
  void setRaw(const uint8_t new_code[], const uint16_t length = kBosch144StateLength);
```